### PR TITLE
Fix.security.vulnerability

### PR DESCRIPTION
**Pull request by joebach71**
upped jsonwebtoken version to resolve security vulnerability.

fix spec - Version 1 can no longer validate encoded token.

Review @heycalmdown @jiho-kr @sejeong @sshtel @Ohen @wokim @resoliwan

### CODE REVIEW

**Comment by resoliwan** [comment]
I am not sure it is safe which upgrading major version of the jsonwebtoken.

It would be better. If you verify us about their is no breaking change between 7 and 8.

File: src/spec/index.spec.ts [hunk 1]
```diff
@@ -15,35 +15,37 @@ function generatePayload() {
...
     it('should treat a binary token as invalid', () => {
         const payload = generatePayload();
         const shrinkedToken = generateJWT(BinaryJWT.shrink(payload, 1));
         const binaryToken = BinaryJWT.encode(shrinkedToken);
         expect(() => jwt.verify(binaryToken, SECRET)).toThrow();
     })
-
-    it('should validate a decoded token', () => {
+    /* Deprecated */
+    xit('should validate a decoded token', () => {
```
Comment (sshtel) [inline src/spec/index.spec.ts]: why is this test disabled?

**Comment by sshtel** [comment]
It would be better with detailed note for fixing security vulnerability

File: src/spec/index.spec.ts [hunk 2]
```diff
@@ -15,35 +15,37 @@ function generatePayload() {
     payload.jti = ObjectId();
     payload.waiting = true;
     payload.types = ['mobile', 'android', 'pub'];
+    payload.exp = Math.floor(Date.now().valueOf() / 1000) + 500 * 60 * 60;
     return payload;
 }
 
 function generateJWT(payload) {
-    return jwt.sign(payload, SECRET, { expiresIn: '500h', noTimestamp: true });
+    return jwt.sign(payload, SECRET, {  noTimestamp: true });
```
Comment (jiho-kr) [inline src/spec/index.spec.ts]: empty space

File: src/spec/index.spec.ts (hunk 1 above)
Comment (sshtel) [reply src/spec/index.spec.ts]: is this test deprecated? then approved

Comment (joebach71) [reply src/spec/index.spec.ts]: jsonwebtoken major update includes removal of JOI validation library which is the security vulnerability.
Change breaks Version 1 encoding/decoding of token which removes exp from token during shrink method.
However, auth island only uses version 2 shrink method which includes exp.

File: src/spec/index.spec.ts (hunk 2 above)
Comment (joebach71) [reply src/spec/index.spec.ts]: removed extra space

**Pull request merged**